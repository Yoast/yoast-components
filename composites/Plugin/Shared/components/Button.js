import React from "react";
import styled from "styled-components";
import flow from "lodash/flow";
import omit from "lodash/omit";
import PropTypes from "prop-types";

import colors from "../../../../style-guide/colors.json";
import SvgIcon from "./SvgIcon";
import { rgba } from "../../../../style-guide/helpers";
import { getRtlStyle } from "../../../../utils/helpers/styled-components";

const settings = {
	minHeight: 32,
	verticalPadding: 4,
	borderWidth: 1,
};

const ieMinHeight = settings.minHeight - ( settings.verticalPadding * 2 ) - ( settings.borderWidth * 2 );

/**
 * Returns a component with applied base button styles.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied base button styles.
 */
export function addBaseStyle( component ) {
	return styled( component )`
		display: inline-flex;
		align-items: center;
		justify-content: center;
		vertical-align: middle;
		border-width: ${ `${ settings.borderWidth }px` };
		border-style: solid;
		margin: 0;
		padding: ${ `${ settings.verticalPadding }px` } 10px;
		border-radius: 3px;
		cursor: pointer;
		box-sizing: border-box;
		font-size: inherit;
		font-family: inherit;
		font-weight: inherit;
		text-align: ${ getRtlStyle( "left", "right" ) };
		overflow: visible;
		min-height: ${ `${ settings.minHeight }px` };

		svg {
			// Safari 10
			align-self: center;
		}

		// Only needed for IE 10+. Don't add spaces within brackets for this to work.
		@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
			::after {
				display: inline-block;
				content: "";
				min-height: ${ `${ ieMinHeight }px` };
			}
		}
	`;
}

/**
 * Returns a component with applied focus styles.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied focus styles.
 */
export function addFocusStyle( component ) {
	return styled( component )`
		&::-moz-focus-inner {
			border-width: 0;
		}

		&:focus {
			outline: none;
			border-color: ${ props => props.focusBorderColor };
			color: ${ props => props.focusColor };
			background-color: ${ props => props.focusBackgroundColor };
			box-shadow: 0 0 3px ${ rgba( colors.$color_blue_dark, .8 ) };
		}
	`;
}

/**
 * Returns a component with applied hover styles.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied hover styles.
 */
export function addHoverStyle( component ) {
	return styled( component )`
		&:hover {
			color: ${ props => props.hoverColor };
			background-color: ${ props => props.hoverBackgroundColor };
			border-color: ${ props => props.hoverBorderColor };
		}
	`;
}

/**
 * Returns a component with applied active styles.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied active styles.
 */
export function addActiveStyle( component ) {
	return styled( component )`
		&:active {
			color: ${ props => props.activeColor };
			background-color: ${ props => props.activeBackgroundColor };
			border-color: ${ props => props.activeBorderColor };
			box-shadow: inset 0 2px 5px -3px ${ rgba( colors.$color_button_border_active, 0.5 ) };
		}
	`;
}

/**
 * Returns a component with applied font size style.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied font size styles.
 */
export function addFontSizeStyles( component ) {
	return styled( component )`
		font-size: 0.8rem;
	`;
}

/**
 * Returns a component with all button selector styles applied.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied styles.
 */
export const addButtonStyles = flow( [
	/*
	 * Styled-components applies the generated CSS classes in a reversed order,
	 * but we want them in the order: base - hover - focus - active.
	 */
	addActiveStyle,
	addFocusStyle,
	addHoverStyle,
	addBaseStyle,
] );

/**
 * Returns a basic styled button.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} styled button.
 */
export const BaseButton = addButtonStyles(
	styled.button`
		color: ${ props => props.textColor };
		border-color: ${ props => props.borderColor };
		background: ${ props => props.backgroundColor };
		box-shadow: 0 1px 0 ${ props => rgba( props.boxShadowColor, 1 ) };
	`
);

BaseButton.propTypes = {
	type: PropTypes.string,
	backgroundColor: PropTypes.string,
	textColor: PropTypes.string,
	borderColor: PropTypes.string,
	boxShadowColor: PropTypes.string,
	hoverColor: PropTypes.string,
	hoverBackgroundColor: PropTypes.string,
	hoverBorderColor: PropTypes.string,
	activeColor: PropTypes.string,
	activeBackgroundColor: PropTypes.string,
	activeBorderColor: PropTypes.string,
	focusColor: PropTypes.string,
	focusBackgroundColor: PropTypes.string,
	focusBorderColor: PropTypes.string,
};

BaseButton.defaultProps = {
	type: "button",
	backgroundColor: colors.$color_button,
	textColor: colors.$color_button_text,
	borderColor: colors.$color_button_border,
	boxShadowColor: colors.$color_button_border,
	hoverColor: colors.$color_button_text_hover,
	hoverBackgroundColor: colors.$color_button_hover,
	hoverBorderColor: colors.$color_button_border_hover,
	activeColor: colors.$color_button_text_hover,
	activeBackgroundColor: colors.$color_button,
	activeBorderColor: colors.$color_button_border_hover,
	focusColor: colors.$color_button_text_hover,
	focusBackgroundColor: colors.$color_white,
	focusBorderColor: colors.$color_blue,
};

/**
 * Returns a styled Button with set font size.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} styled button.
 */
export const Button = addFontSizeStyles( BaseButton );

/**
 * Applies styles to SvgIcon for IconButton with text.
 *
 * @param {ReactElement} icon The original SvgIcon.
 *
 * @returns {ReactElement} SvgIcon with text styles applied.
 */
function addIconTextStyle( icon ) {
	return styled( icon )`
		margin: ${ getRtlStyle( "0 8px 0 0", "0 0 0 8px" ) };
		flex-shrink: 0;
	`;
}

/**
 * Returns an icon button that can optionally contain text.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} styled icon button.
 */
export const IconButton = ( props ) => {
	const { children: text, icon, iconColor } = props;

	let IconComponent = SvgIcon;
	if ( text ) {
		IconComponent = addIconTextStyle( IconComponent );
	}

	const newProps = omit( props, "icon" );

	return (
		<Button { ...newProps }>
			<IconComponent icon={ icon } color={ iconColor } />
			{ text }
		</Button>
	);
};

IconButton.propTypes = {
	icon: PropTypes.string.isRequired,
	iconColor: PropTypes.string,
	children: PropTypes.oneOfType( [
		PropTypes.arrayOf( PropTypes.node ),
		PropTypes.node,
		PropTypes.string,
	] ),
};

IconButton.defaultProps = {
	iconColor: "#000",
};

/**
 * Returns an icons button that can optionally contain a prefix and / or a suffix icon.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} styled icon button.
 */
export const IconsButton = ( props ) => {
	const {
		children,
		className,
		prefixIcon,
		suffixIcon,
		...buttonProps
	} = props;

	return (
		<Button className={ className } { ...buttonProps }>
			{ prefixIcon &&
				<SvgIcon
					icon={ prefixIcon.icon }
					color={ prefixIcon.color }
					size={ prefixIcon.size }
				/>
			}
			{ children }
			{ suffixIcon &&
				<SvgIcon
					icon={ suffixIcon.icon }
					color={ suffixIcon.color }
					size={ suffixIcon.size }
				/>
			}
		</Button>
	);
};

IconsButton.propTypes = {
	className: PropTypes.string,
	prefixIcon: PropTypes.shape( {
		icon: PropTypes.string,
		color: PropTypes.string,
		size: PropTypes.string,
	} ),
	suffixIcon: PropTypes.shape( {
		icon: PropTypes.string,
		color: PropTypes.string,
		size: PropTypes.string,
	} ),
	children: PropTypes.oneOfType( [
		PropTypes.arrayOf( PropTypes.node ),
		PropTypes.node,
		PropTypes.string,
	] ),
};

IconsButton.defaultProps = {
	prefixIcon: {
		icon: "",
		color: colors.$black,
		size: "16px",
	},
	suffixIcon: {
		icon: "",
		color: colors.$black,
		size: "16px",
	},
};
