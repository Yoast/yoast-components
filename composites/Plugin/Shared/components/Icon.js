import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import _omit from "lodash/omit";

/**
 * Returns the Icon component.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} Icon component.
 */
export const Icon = ( props ) => {
	const IconComponent = styled( props.icon )`
		width: ${ props.size };
		height: ${ props.size };
		fill: ${ props.color };
		flex: none;
	`;

	// Remove the props that are no longer needed
	const newProps = _omit( props, [ "icon", "size", "color" ] );

	return <IconComponent role="img" aria-hidden="true" focusable="false" { ...newProps } />;
};

Icon.propTypes = {
	icon: PropTypes.func.isRequired,
	color: PropTypes.string.isRequired,
	size: PropTypes.string,
};

Icon.defaultProps = {
	size: "16px",
};

// Safari doesn't expose the ARIA role set on SVG element to browsers.
export const IconWithAriaLabel = ( props ) => {
	return <span role="img" aria-label={ props.ariaLabel }>
		<Icon
			icon={ props.icon }
			color={ props.color }
			size={ props.size }
			role="presentation"
		/>
	</span>;
};

IconWithAriaLabel.propTypes = {
	icon: PropTypes.func.isRequired,
	color: PropTypes.string.isRequired,
	size: PropTypes.string.isRequired,
	ariaLabel: PropTypes.string.isRequired,
};
