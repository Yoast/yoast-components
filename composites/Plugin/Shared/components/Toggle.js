import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import colors from "../../../../style-guide/colors.json";
import { __ } from "@wordpress/i18n";

const ToggleBar = styled.div`
	background-color: ${ props => props.isEnabled ? "#a5d6a7" : colors.$color_button_border };
	border-radius: 7px;
	height: 14px;
	width: 30px;
	cursor: pointer;
	margin: 0;
	outline: 0;
	&:focus > span {
		box-shadow: inset 0 0 0 1px ${colors.$color_white}, 0 0 0 1px #5b9dd9, 0 0 2px 1px rgba(30, 140, 190, .8);
	}
`;

const ToggleBullet = styled.span`
	background-color: ${ props => props.isEnabled ? colors.$color_green_medium_light : colors.$color_grey_medium_dark };
	margin-left: ${ props => props.isEnabled ? "12px" : "-2px" };
	box-shadow: 0 2px 2px 2px rgba(0, 0, 0, 0.1);
	border-radius: 100%;
	height: 20px;
	width: 20px;
	position: absolute;
	margin-top: -3px;
`;

const ToggleVisualLabel = styled.span`
	font-size: 14px;
	line-height: 20px;
	width: 30px;
	text-align: center;
	display: inline-block;
	margin: 0;
	font-style: italic;
`;

const ToggleDiv = styled.div`
	display: flex;
	width: 100%;
	justify-content: space-between;
	align-items: center;
`;

class Toggle extends React.Component {
	/**
	 * Sets the toggle object.
	 *
	 * @param {Object} props The props to use.
	 *
	 * @returns {void}
	 */
	constructor( props ) {
		super( props );

		this.onClick = this.props.onToggleDisabled;

		this.setToggleState = this.setToggleState.bind( this );

		if ( props.disable !== true ) {
			this.onClick = this.setToggleState.bind( this );
		}
	}

	/**
	 * Returns the rendered HTML.
	 *
	 * @returns {ReactElement} The rendered HTML.
	 */
	render() {
		return(
			<ToggleDiv>
				{ /* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions */ }
				<label
					htmlFor={ this.props.id }
					onClick={ this.onClick }
				>
					{ this.props.labelText }
				</label>
				<ToggleBar
					isEnabled={ this.props.isEnabled }
					onClick={ this.onClick }
					onKeyDown={ this.setToggleState }
					tabIndex="0"
					role="checkbox"
					aria-label={ this.props.ariaLabel }
					aria-checked={ this.props.isEnabled }
					id={ this.props.id }
				>
					<ToggleBullet isEnabled={ this.props.isEnabled } />
				</ToggleBar>
				<ToggleVisualLabel aria-hidden="true">
					{ this.props.isEnabled ? __( "On", "yoast-components" ) : __( "Off", "yoast-components" ) }
				</ToggleVisualLabel>
			</ToggleDiv>
		);
	}

	/**
	 * Sets the state to the opposite of the current state.
	 *
	 * @param {Object} event React SyntheticEvent.
	 *
	 * @returns {void}
	 */
	setToggleState( event ) {
		// Makes the toggle actionable with the Space bar key.
		if ( event.type === "keydown" && event.keyCode !== 32 ) {
			return;
		}

		this.props.onSetToggleState( ! this.props.isEnabled );
	}
}

Toggle.propTypes = {
	isEnabled: PropTypes.bool,
	ariaLabel: PropTypes.string.isRequired,
	onSetToggleState: PropTypes.func,
	disable: PropTypes.bool,
	onToggleDisabled: PropTypes.func,
	id: PropTypes.string.isRequired,
	labelText: PropTypes.string,
};

Toggle.defaultProps = {
	isEnabled: false,
	onSetToggleState: () => {},
	disable: false,
	labelText: "",
};

export default Toggle;
