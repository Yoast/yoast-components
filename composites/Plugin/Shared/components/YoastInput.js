// External dependencies.
import styled from "styled-components";

// Internal dependencies.
import colors from "../../../../style-guide/colors.json";

export const YoastInputContainer = styled.div`
	display: flex;
	flex-direction: column;
	margin: 1em 0;
`;

export const YoastInputLabel = styled.label`
	font-size: 1em;
	font-weight: bold;
	margin-bottom: 0.5em;
	display: block;
`;

export const YoastInputField = styled.input`
	border: 1px solid ${ colors.$color_input_border };
	padding: 0.75em;
	font-size: 1em;
`;

export const YoastInputButtonContainer = styled.div`
	border: 1px solid ${ colors.$color_input_border };
	display: flex;
	flex-direction: row;
`;
