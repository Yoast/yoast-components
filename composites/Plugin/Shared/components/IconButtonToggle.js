import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import colors from "../../../../style-guide/colors.json";
import { Icon } from "./Icon";
import { rgba } from "../../../../style-guide/helpers";

const ChangingIconButtonBase = styled.button`
	box-sizing: border-box;
	min-width: 32px;
	display: inline-block;
	border: 1px solid ${ colors.$color_button_border };
	background-color: ${ props => props.pressed ? props.pressedBackground : props.unpressedBackground };
	box-shadow: ${ props => props.pressed
		? `inset 0 2px 0 ${ rgba( props.pressedBoxShadowColor, 0.7 ) }`
		: `0 1px 0 ${ rgba( props.unpressedBoxShadowColor, 0.7 ) }` };
	border-radius: 3px;
	cursor: pointer;
	padding: 0;
	height: ${ props => props.pressed ? "24px" : "25px" };
	
	&:hover {
		border-color: ${ props => props.hoverBorderColor }
	}
`;

const ChangingIconButton = ( props ) => {
	return (
		<ChangingIconButtonBase
			onClick={ props.onClick }
			pressed={ props.pressed }
			unpressedBoxShadowColor={ props.unpressedBoxShadowColor }
			pressedBoxShadowColor={ props.pressedBoxShadowColor }
			pressedBackground={ props.pressedBackground }
			unpressedBackground={ props.unpressedBackground }
			id={ props.id }
			aria-label={ props.ariaLabel }
			aria-pressed={ props.pressed }
			unpressedIconColor={ props.unpressedIconColor }
			pressedIconColor={ props.pressedIconColor }
			hoverBorderColor={ props.hoverBorderColor }
		>
			{ props.pressed
				? <Icon
					icon={ props.icon }
					color={ props.pressedIconColor }
					size="18px"/>
				: <Icon
					icon={ props.icon }
					color={ props.unpressedIconColor }
					size="18px"/>
			}
		</ChangingIconButtonBase>
	);
};

ChangingIconButton.propTypes = {
	id: PropTypes.string.isRequired,
	ariaLabel: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
	boxShadowColor: PropTypes.string,
	unpressedBoxShadowColor: PropTypes.string,
	pressedBoxShadowColor: PropTypes.string,
	pressedBackground: PropTypes.string,
	unpressedBackground: PropTypes.string,
	pressedIconColor: PropTypes.string,
	unpressedIconColor: PropTypes.string,
	icon: PropTypes.func.isRequired,
	pressed: PropTypes.bool.isRequired,
	hoverBorderColor: PropTypes.string,
};

ChangingIconButton.defaultProps = {
	unpressedBoxShadowColor: colors.$color_button_border,
	pressedBoxShadowColor: colors.$color_purple,
	pressedBackground: colors.$color_pink_dark,
	unpressedBackground: colors.$color_button,
	pressedIconColor: colors.$color_white,
	unpressedIconColor: colors.$color_button_text,
	hoverBorderColor: colors.$color_white,
};

export default ChangingIconButton;
