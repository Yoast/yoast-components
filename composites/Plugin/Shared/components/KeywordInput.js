// External dependencies.
import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import uniqueId from "lodash/uniqueId";
import { __ } from "@wordpress/i18n";
import noop from "lodash/noop";

// Internal dependencies.
import colors from "../../../../style-guide/colors.json";
import { YoastInputButtonContainer, YoastInputField } from "./YoastInput";
import SvgIcon from "./SvgIcon";

const errorColor = colors.$color_red;
const greyColor = colors.$color_grey_medium;

const KeywordInputContainer = styled.div`
	display: flex;
	flex-direction: column;
	margin: 1em 0;
`;

const KeywordFieldLabel = styled.label`
	font-size: 1em;
	font-weight: bold;
	margin-bottom: 0.5em;
`;

const KeywordField = styled( YoastInputField )`
	flex: 1 !important;
	border: none !important;
	box-shadow: none !important;

	&.hasError {
		border-color: ${ errorColor };
		outline-color: ${ errorColor };
	}
`;

const ErrorText = styled.p`
	color: ${ errorColor };
	margin: 0.5em 0 0 0;
	min-height: 1.8em;
`;

const BorderlessButton = styled.button`
	border: none;
	box-shadow: none;
	background: none;
`;

const RemoveIcon = styled( SvgIcon )`
	margin-top: 4px;
	cursor: pointer;
`;

class KeywordInput extends React.Component {
	/**
	 * Constructs a KeywordInput component.
	 *
	 * @param {Object}   props           The props for the KeywordInput.
	 * @param {string}   props.id        The id of the KeywordInput.
	 * @param {string}   props.label     The label of the KeywordInput.
	 * @param {boolean}  props.showLabel Toggle between an actual label or an aria-label on the input.
	 * @param {string}   props.keyword   The initial keyword passed to the state.
	 * @param {Function} props.onChange  The function that is triggered when the keyword input field is changed.
	 *
	 * @returns {void}
	 */
	constructor( props ) {
		super( props );

		this.handleChange = this.handleChange.bind( this );
		this.displayErrorMessage = this.displayErrorMessage.bind( this );
	}

	/**
	 * Checks the keyword input for comma-separated words.
	 *
	 * @param {string} keywordText The text of the input.
	 *
	 * @returns {boolean} Returns true if a comma was found.
	 */
	checkKeywordInput( keywordText ) {
		return keywordText.includes( "," );
	}

	/**
	 * Displays the error message
	 *
	 * @param {boolean} showErrorMessage Whether or not the error message has to be shown.
	 *
	 * @returns {ReactElement} ErrorText The error message element.
	 */
	displayErrorMessage( showErrorMessage ) {
		if ( showErrorMessage && this.props.keyword !== "" ) {
			return (
				<ErrorText role="alert">
					{ __( "Are you trying to use multiple keywords? You should add them separately below.", "yoast-components" ) }
				</ErrorText>
			);
		}
	}

	/**
	 * Handles changes in the KeywordInput, sets the state if a change has been made.
	 *
	 * @param {SyntheticEvent} event The onChange event.
	 *
	 * @returns {void}
	 */
	handleChange( event ) {
		this.props.onChange( event.target.value );
	}

	/**
	 * Renders an input field, a label, and if the condition is met, an error message.
	 *
	 * @returns {ReactElement} The KeywordField react component including its label and eventual error message.
	 */
	render() {
		const { id, showLabel, keyword, onRemoveKeyword } = this.props;
		const showErrorMessage = this.checkKeywordInput( keyword );

		const label = __( "Focus keyword:", "yoast-components" );

		// The aria label should not be shown if there is a visible label.
		const showAriaLabel = ! showLabel;

		return(
			<KeywordInputContainer>
				{ showLabel && <KeywordFieldLabel htmlFor={ id }>
					{ label }
				</KeywordFieldLabel> }
				<YoastInputButtonContainer>
					<KeywordField
						aria-label={ showAriaLabel ? label : null }
						type="text"
						id={ id }
						className={ showErrorMessage ? "hasError" : null }
						onChange={ this.handleChange }
						value={ keyword }
					/>
					{ onRemoveKeyword !== noop && (
						<BorderlessButton onClick={ onRemoveKeyword } >
							<RemoveIcon
								size="18px"
								icon="times-circle"
								color={ greyColor }
							/>
						</BorderlessButton>
					) }
				</YoastInputButtonContainer>
				{ this.displayErrorMessage( showErrorMessage ) }
			</KeywordInputContainer>
		);
	}
}

KeywordInput.propTypes = {
	id: PropTypes.string,
	showLabel: PropTypes.bool,
	keyword: PropTypes.string,
	onChange: PropTypes.func.isRequired,
	onRemoveKeyword: PropTypes.func,
};

KeywordInput.defaultProps = {
	id: uniqueId( "yoast-keyword-input-" ),
	showLabel: true,
	keyword: "",
	onRemoveKeyword: noop,
};

export default KeywordInput;
