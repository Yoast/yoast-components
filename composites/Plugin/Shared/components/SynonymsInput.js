import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import uniqueId from "lodash/uniqueId";
import { YoastInputContainer, YoastInputField, YoastInputLabel } from "./YoastInput";

const ExplanationText = styled.p`
	font-size: .8em;
`;

const SynonymsInput = ( { id, label, value, onChange, explanationText } ) => {
	return(
		<YoastInputContainer>
			<YoastInputLabel htmlFor={ id }>
				{ label }
			</YoastInputLabel>
			{ explanationText !== "" && <ExplanationText>
				{ explanationText }
			</ExplanationText> }
			<YoastInputField
				aria-label={ label }
				type="text"
				id={ id }
				onChange={ onChange }
				value={ value }
			/>
		</YoastInputContainer>
	);
};

SynonymsInput.propTypes = {
	id: PropTypes.string,
	label: PropTypes.string,
	value: PropTypes.string,
	onChange: PropTypes.func.isRequired,
	explanationText: PropTypes.string,
};

SynonymsInput.defaultProps = {
	id: uniqueId( "synonyms-input-" ),
	label: "",
	value: "",
	explanationText: "",
};

export default SynonymsInput;
