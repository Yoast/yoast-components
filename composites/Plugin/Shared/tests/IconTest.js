import React from "react";
import renderer from "react-test-renderer";

import edit from "../../../../style-guide/svg/edit.svg";
import { Icon } from "../components/Icon";
import { IconWithAriaLabel } from "../components/Icon";

test( "the Icon matches the snapshot", () => {
	const component = renderer.create(
		<Icon icon={ edit } color="black" size="16px" />
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the IconWithAriaLabel matches the snapshot", () => {
	const component = renderer.create(
		<IconWithAriaLabel icon={ edit } color="black" size="16px" ariaLabel="This is an aria label" />
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
