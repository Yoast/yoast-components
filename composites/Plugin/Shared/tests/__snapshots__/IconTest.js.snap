// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Icon matches the snapshot 1`] = `
<svg
  aria-hidden="true"
  className="sc-bdVaJa jCwthi"
  focusable="false"
  role="img"
/>
`;

exports[`the IconWithAriaLabel matches the snapshot 1`] = `
<span
  aria-label="This is an aria label"
  role="img"
>
  <svg
    aria-hidden="true"
    className="sc-bwzfXH fWnAvQ"
    focusable="false"
    role="presentation"
  />
</span>
`;
